### PR TITLE
boards/stm32-based: use shared configuration snippets

### PR DESCRIPTION
**Pull request by haukepetersen**
### Contribution description
This is another change cut from #8044 to reduce the vast code duplication in the (STM32) board configurations.

Presented in this PR in my idea for using shared configuration snippets in the board's `periph_conf.h` files. This is demonstrated so far only for some selected `stm32f4`-based boards, and focused on their clock configuration only. Once we agree for this approach, we can optimize the configuration for more STM32-based boards, in small, easily revieweable steps...

### Issues/PRs references
continuation of merging #8044 in small steps...

### CODE REVIEW

File: boards/nucleo144-f429/include/periph_conf.h [hunk 1]
```diff
@@ -22,42 +22,12 @@
 #define PERIPH_CONF_H
 
 #include "periph_cpu.h"
+#include "f4/cfg_clock_168_8_1.h"
```
Comment (vincent-d) [inline boards/nucleo144-f429/include/periph_conf.h]: missing `#include "cfg_spi_divtable.h"`

Comment (haukepetersen) [reply boards/nucleo144-f429/include/periph_conf.h]: ups, will fix

**Comment by haukepetersen** [comment]
fixed missing include

**Comment by haukepetersen** [comment]
also fixed header guards

**Comment by haukepetersen** [comment]
rebased

**Comment by haukepetersen** [comment]
and squashed...

File: boards/common/stm32/include/cfg_spi_divtable.h [hunk 2]
```diff
@@ -0,0 +1,266 @@
...
+ * @ingroup     boards_common_stm32
+ * @{
+ *
+ * @file
+ * @brief       Collection of pre-computed bus pre-scalers for SPI configuration
+ *
+ * @author      Hauke Petersen <hauke.petersen@fu-berlin.de>
+ */
+
+#ifndef PERIPH_CFG_SPI_DIVTABLE_H
```
Comment (vincent-d) [inline boards/common/stm32/include/cfg_spi_divtable.h]: This header guard is still wrong according to Murdock

**Comment by haukepetersen** [comment]
fixed the header guard issue (squashed right in) and rebased.

**Comment by haukepetersen** [comment]
@vincent-d would you mind to give this one last look and press the big green button?! Thanks!

**Comment by vincent-d** [comment]
I'm OK to merge if nobody objects (is there any potential conflict with i2c refactoring?) @aabadie @dylad ?

**Comment by dylad** [comment]
It may introduces some conflicts with the I2C refactoring. @aabadie will be able to confirm it better than me.

**Comment by aabadie** [comment]
> is there any potential conflict with i2c refactoring?

I don't know, it needs to be confirmed locally first I would say.

**Comment by vincent-d** [comment]
I tried to merge #9202 and this one in master and there is no conflicts, so I guess we can merge. @aabadie ?

**Comment by aabadie** [comment]
> I tried to merge #9202 and this one in master and there is no conflicts, so I guess we can merge. 

Let's go then :)

**Pull request merged**